File: r/raindrops/raindrops.R
raindrops <- function(number) {
    res <- ""

    if (number %% 3 == 0) {
        res <- paste(res, "Pling", sep="")
    }

    if (number %% 5 == 0) {
        res <- paste(res, "Plang", sep="")
    }

    if (number %% 7 == 0) {
        res <- paste(res, "Plong", sep="")
    }


    if (res == "") {
        res <- as.character(number)
    }

    return (res)
}

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n
    | converted == "" = show n
    | otherwise = converted
    where
        converted = concatMap mappingFunction numberMapping
        mappingFunction (number, mapping) = if n `mod` number == 0 then mapping else ""
        numberMapping = [(3, "Pling"), (5, "Plang"), (7, "Plong")]

